features/fixtures/mazerunner/src/main/java/com/bugsnag/android/mazerunner/scenarios/AutoContextScenario.kt

```diff
@@ -18,12 +20,11 @@ internal class AutoContextScenario(config: Configuration,
 
     override fun run() {
         super.run()
+        registerActivityLifecycleCallbacks()
         context.startActivity(Intent(context, SecondActivity::class.java))
-
-        val a = context as Activity
-        a.window.decorView.postDelayed({
-            Bugsnag.notify(generateException())
-        }, 2000)
     }
 
+    override fun onActivityStarted(activity: Activity) {
```

Comment:
This will be invoked as soon as the activity is in the foreground. The context should be automatically set to the activity's name at this time, so there's no point waiting for 2s.